packages/web/src/lib/web-mode.ts

```diff
@@ -269,4 +269,46 @@ export class WebMode {
 		}
 		throw new Error("Transaction Error");
 	}
+
+	async makeTx(transactions: ExecParams[]): Promise<Transaction[]> {
+		const txns: Transaction[] = [];
+		for (const [_, txn] of transactions.entries()) {
+			txns.push(mkTransaction(txn, await this.getSuggestedParams(txn.payFlags)));
+		}
+		return txns;
+	}
+	async signTx(transaction: algosdk.Transaction): Promise<SignedTransaction> {
+		const binaryTx = transaction.toByte();
+		const base64Tx = this.algoSigner.encoding.msgpackToBase64(binaryTx);
+		const signedTx = await this.algoSigner.signTxn([
+			{
+				txn: base64Tx,
+			},
+		]);
+		const signedTxJson = signedTx as JsonPayload;
+		const blob = signedTxJson.blob as string;
+		const blobArray = Uint8Array.from(Buffer.from(blob, "base64"));
+		return algosdk.decodeSignedTransaction(blobArray);
+	}
+	async makeAndSignTx(transactions: ExecParams[]): Promise<SignedTransaction[]> {
+		const signedTxns: SignedTransaction[] = [];
+		const txns: Transaction[] = await this.makeTx(transactions);
+		txns.forEach(async (txn) => signedTxns.push(await this.signTx(txn)));
+		return signedTxns;
+	}
+	async sendTxAndWait(
+		txs: SignedTransaction[],
+		rounds?: number
+	): Promise<algosdk.modelsv2.PendingTransactionResponse> {
+		if (txs.length < 1) {
+			throw Error("No transactions to process");
+		} else {
+			const txInfo = await this.sendGroupTransaction(txs);
+
+			if (txInfo && typeof txInfo.txId === "string") {
```

Comment:
why checking `txId`?

Reply:
this is how it was already implemented in web-mode. I was following the implementations of already existing functions